### DISCUSSION TITLE
poetry build - whl does not include optional dependency groups

### DISCUSSION THREAD

**Discussion by TomekPro** [python-poetry/poetry#9788] [General]
Hi,
I have a pyproject.toml with a few dependency groups. It works like a charm with poetry install, but when I want to create .whl out of a environment it keeps inside only "main" dependencies - not optional ones. Any idea how to handle it? I don't want to keep several .toml files.

Example of pytproject.toml:
```
[tool.poetry]
name = "ml_tools"
version = "0.1"
description = "ML toolbox repo"

[tool.poetry.dependencies]
python = "3.11.8"
pandas = "1.5.1"


[tool.poetry.group.optional_group_1]
optional = true

[tool.poetry.group.optional_group_1.dependencies]
torch = {url = "https://download.pytorch.org/whl/cpu/torch-2.1.1%2Bcpu-cp311-cp311-linux_x86_64.whl"}


[tool.poetry.group.dev]
optional = true

[tool.poetry.group.dev.dependencies]
dvc = {extras = ["s3", "gs"], version = "2.58.1"}

[build-system]
requires = ["setuptools", "poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"

```

**Comment by Secrus** [comment]
Dependency groups are dev dependencies, if you want to include those in the wheel, they should be your main dependencies.

**Reply by boba-fett-hunter** [reply]
Not necessarily - depending on the need, I may want to build whl with different dependencies. And use only part of the code contained in the tools repo - related to dependencies.

**Reply by TomekPro** [reply]
Yeah, and there is no a way to make them main? Maybe some other way I can keep one .toml and have many optional groups/flavours of it that would be reflected when building a whl?
In make case I have a ml_toolbox repo, and I don't want to install all its dependencies in other repos that use this toolbox, just selected,necessary groups.

**Reply by boba-fett-hunter** [reply]
Something like `poetry build --with` would be useful, as in `poetry install --with`. But I didn't find anything about such a feature in the documentation.

**Reply by Secrus** [reply]
That would mean you would have different metadata for wheels within the same projects, which is not something that should be done.

**Reply by pramitchoudhary** [reply]
There could be scenarios where this setup could help in avoiding to maintain version of `pyproject.toml`.
Also, if we think `install` and `build` syntax should shadow each other?
